internal/server/server.go

```diff
@@ -551,6 +553,35 @@ func (s *Server) VerifyIdentityHeader(nextHandler echo.HandlerFunc) echo.Handler
 	return func(ctx echo.Context) error {
 		request := ctx.Request()
 
+		loader := openapi3.NewSwaggerLoader()
+		doc, err := loader.LoadSwaggerFromFile("api.yaml")
+		if err != nil {
+			panic(err)
+		}
+
+		if err := doc.Validate(loader.Context); err != nil {
+			panic(err)
+		}
+
+		router, err := legacyrouter.NewRouter(doc)
+		if err != nil {
+			panic(err)
+		}
+		route, params, err := router.FindRoute(request)
```

Comment:
This should move to a separate middleware imo. Which is slotted in after the  `VerifyIdentityHeader` middleware.